server/src/main/java/org/apache/iotdb/db/metadata/cache/DataNodeSchemaCache.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.metadata.cache;
+
+import org.apache.iotdb.commons.utils.TestOnly;
+import org.apache.iotdb.db.conf.IoTDBConfig;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.metadata.path.AlignedPath;
+import org.apache.iotdb.db.metadata.path.MeasurementPath;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.mpp.common.schematree.PathPatternTree;
+import org.apache.iotdb.db.mpp.common.schematree.SchemaTree;
+import org.apache.iotdb.db.mpp.sql.analyze.FakeSchemaFetcherImpl;
+import org.apache.iotdb.db.mpp.sql.analyze.ISchemaFetcher;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.write.schema.IMeasurementSchema;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class takes the responsibility of metadata cache management of all DataRegions under
+ * StorageEngine
+ */
+public class DataNodeSchemaCache {
+  private static final Logger logger = LoggerFactory.getLogger(DataNodeSchemaCache.class);
+
+  private static IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+
+  private Cache<PartialPath, SchemaCacheEntity> schemaEntityCache;
```

Comment:
Could we use MTree directly?

Reply:
There's no cache eviction adapt to MTree, that is why we use existing map cache implement. We can implement a TreeCache in future work, which could be used in many scenarios related to tree structure and prefix compress.

Reply:
The design intention for this cache is, for insert operation, they just need to provide me a partialPath key and then I will tell then a related schemaCacheEntity. If they want to auto create schema, the partialPath should be MeasurementPath or AlignedPath which should contain some schema info. So it seems that current MTree structure is kind of redundancy because here we do not need such a tree structure.